### PR TITLE
implement deeplink gradle plugin

### LINKED ISSUE #251: Deep link support: Gradle plugin that generated intent filters
Based on a toml file that lists all supported deep links automatically generate the intent filters needed to support them on the launcher activity.

See for how to modify the manifest from a task: https://github.com/android/gradle-recipes/blob/agp-7.4/BuildSrc/manifestUpdaterTest/buildSrc/src/main/kotlin/ManifestTransformerTask.kt

### PR DESCRIPTION
**Pull request by gabrielittner**
Implements the remaining parts of the plugin to generate deep links from a TOML file.

Closes #251

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -93,6 +93,8 @@ androidx-savedstate = { module = "androidx.savedstate:savedstate", version.ref =
 androidx-viewbinding = { module = "androidx.databinding:viewbinding", version.ref = "android-gradle" }
 
 uri = { module = "com.eygraber:uri-kmp", version.ref = "uri" }
+toml = { module = "net.peanuuutz.tomlkt:tomlkt", version = "0.3.5" }
```
Comment (gabrielittner) [inline gradle/libs.versions.toml:96]: Found another library that is based on kotlinx.serialization, published to maven central and supports all toml features.

File: deeplinks-plugin/src/main/kotlin/com/freeletics/khonshu/deeplinks/plugin/DeepLinkDefinitions.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        }
+    }
+}
+
+@Serializable
+public data class DeepLinkDefinition(
+    val patterns: List<PatternDefinition>,
+    val prefixes: List<PrefixDefinition> = emptyList(), // use global if empty
+    val placeholders: List<PlaceholderDefinition> = emptyList(), // use global if empty or if key not found
+    val exampleQueries: List<String> = emptyList(),
```
Comment (gabrielittner) [inline deeplinks-plugin/src/main/kotlin/com/freeletics/khonshu/deeplinks/plugin/DeepLinkDefinitions.kt:28]: This is only here for our list/test script. Currently we have deep links with query parameters defined as separate deep links which doesn't work well. This makes it possible to define query parameters here that can be used in examples.

File: navigation/src/commonMain/kotlin/com/freeletics/khonshu/navigation/deeplinks/DeepLinkHandler.kt [hunk 3]
```diff
@@ -64,11 +64,11 @@ public interface DeepLinkHandler {
...
         init {
             check(!value.startsWith("/")) { "Pattern should not start with a / but is $value" }
-            check(!value.contains("?")) { "Pattern should not contain any query parameters is $value" }
+            check(!value.contains("?")) { "Pattern should not contain any query parameters but is $value" }
         }
     }
 
     private companion object {
-        private val PREFIX_REGEX = "[a-z]+://[a-z0-9._-]+.[a-z]+".toRegex()
+        private val PREFIX_REGEX = "^[a-z]+://[a-z0-9._-]+\\.[a-z]+$".toRegex()
```
Comment (gabrielittner) [inline navigation/src/commonMain/kotlin/com/freeletics/khonshu/navigation/deeplinks/DeepLinkHandler.kt:72]: This allowed partial matches before.

**Review by kboyarshinov — APPROVED** [review]
Nice 👍 🎉

**Pull request merged**